volume-providers/vpc/vpcclient/instances/iks_attach_volume.go

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * IBM Confidential
+ * OCO Source Materials
+ * IBM Cloud Container Service, 5737-D43
+ * (C) Copyright IBM Corp. 2018, 2019 All Rights Reserved.
+ * The source code for this program is not  published or otherwise divested of
+ * its trade secrets, irrespective of what has been deposited with
+ * the U.S. Copyright Office.
+ ******************************************************************************/
+
+package instances
+
+import (
+	"github.com/IBM/ibmcloud-storage-volume-lib/lib/utils"
+	"github.com/IBM/ibmcloud-storage-volume-lib/volume-providers/vpc/vpcclient/client"
+	"github.com/IBM/ibmcloud-storage-volume-lib/volume-providers/vpc/vpcclient/models"
+	"go.uber.org/zap"
+	"time"
+)
+
+// AttachVolume attached volume to instances with givne volume attachment details
+func (vs *IKSVolumeAttachService) AttachVolume(volumeAttachmentTemplate *models.VolumeAttachment, ctxLogger *zap.Logger) (*models.VolumeAttachment, error) {
+	defer util.TimeTracker("IKS AttachVolume", time.Now())
+
+	operation := &client.Operation{
+		Name:   "AttachVolume",
+		Method: "POST",
+	}
+	operation.PathPattern = vs.pathPrefix + "createAttachment"
+
+	var volumeAttachment models.VolumeAttachment
+	apiErr := vs.receiverError
+
+	request := vs.client.NewRequest(operation)
+
+	request = request.SetQueryValue(IksClusterQueryKey, *volumeAttachmentTemplate.ClusterID)
+	request = request.SetQueryValue(IksWorkerQueryKey, *volumeAttachmentTemplate.InstanceID)
+	vol := *volumeAttachmentTemplate.Volume
+	request = request.SetQueryValue(IksVolumeQueryKey, vol.ID)
+
+	ctxLogger.Info("Equivalent curl command and query parameters", zap.Reflect("URL", request.URL()), zap.Reflect("Payload", volumeAttachmentTemplate), zap.Reflect("Operation", operation), zap.Reflect(IksClusterQueryKey, volumeAttachmentTemplate.InstanceID), zap.Reflect(IksWorkerQueryKey, volumeAttachmentTemplate.InstanceID), zap.Reflect(IksVolumeQueryKey, vol.ID))
+
+	_, err := request.JSONBody(volumeAttachmentTemplate).JSONSuccess(&volumeAttachment).JSONError(apiErr).Invoke()
```

Comment:
we can directly use `vs.receiverError` instead of declaring  `apiErr `

Reply:
applicable for other places

Reply:
We have set it to the same.

apiErr := vs.receiverError

https://github.com/IBM/ibmcloud-storage-volume-lib/pull/190/files#diff-f9a7374775ebfedaaad971ec340b6a86R32